server/src/controllers/auth.controller.js

```
@@ -0,0 +1,167 @@
+import { ApiError } from '../exceptions/api.error.js';
+import { Token } from '../models/token.js';
+import { User } from '../models/user.js';
+import { validateEmail, validatePassword } from '../utils/validation.js';
+import { jwtService } from '../services/jwt.service.js';
+import { tokenService } from '../services/token.service.js';
+import { userService } from '../services/user.service.js';
+import bcrypt from 'bcrypt';
+
+const ACTIVATED_STATUS = 'activated';
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    name: !name ? 'Name is required' : null,
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.name || errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPass = await bcrypt.hash(password, 5);
+
+  await userService.register(name, email, hashedPass);
+
+  res.send({ message: 'Check your email' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    throw ApiError.badRequest();
+  }
+
+  user.activationToken = ACTIVATED_STATUS;
+  await user.save();
+
+  await generateTokens(res, user);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  }
+
+  await generateTokens(res, user);
```

The task requires that if a user is not active, you should prevent them from logging in and ask them to activate their email. This implementation is missing a check for the user's activation status. You should add a condition here to verify if `user.activationToken` is `'activated'` before generating tokens. If it is not, you should throw an appropriate error, like `ApiError.notActivated()`.